libs/orchestrator-ui-components/src/lib/components/Table/DataGridTable.tsx

```diff
@@ -0,0 +1,140 @@
+import {
+    EuiDataGrid,
+    EuiDataGridCellValueElementProps,
+    EuiDataGridStyle,
+} from '@elastic/eui';
+import { useRef, useState } from 'react';
+import {
+    columnSortToEuiDataGridSorting,
+    ControlColumn,
+    getInitialColumnOrder,
+    DataGridTableColumns,
+} from './dataGridColumns';
+import {
+    EuiDataGridControlColumn,
+    EuiDataGridPaginationProps,
+} from '@elastic/eui/src/components/datagrid/data_grid_types';
+import { DataSorting } from './columns';
+
+// Total height of grid button bar, table header and pagination bar
+const EUI_DATA_GRID_HEIGHT_OFFSET = 103;
+
+const GRID_STYLE: EuiDataGridStyle = {
+    border: 'horizontal',
+    stripes: false,
+    rowHover: 'highlight',
+    header: 'shade',
+    cellPadding: 'l',
+    fontSize: 'm',
+    footer: 'overline',
+};
+
+export type Pagination = EuiDataGridPaginationProps & {
+    totalRecords: number;
+};
+
+export type DataGridTableProps<T> = {
+    data: T[];
+    pagination: Pagination;
+    columns: DataGridTableColumns<T>;
+    leadingControlColumns?: ControlColumn<T>[];
+    trailingControlColumns?: ControlColumn<T>[];
+    initialColumnOrder: Array<keyof T>;
+    dataSorting?: DataSorting<T>;
+    handleRowClick?: (row: T) => void;
+    updateDataSorting?: (updatedDataSorting: DataSorting<T>) => void;
+};
+
+export const DataGridTable = <T,>({
+    data,
+    pagination,
+    columns,
+    leadingControlColumns,
+    trailingControlColumns,
+    initialColumnOrder,
+    dataSorting,
+    handleRowClick,
+    updateDataSorting,
+}: DataGridTableProps<T>) => {
+    const initialColumnOrderRef = useRef(
+        getInitialColumnOrder(columns, initialColumnOrder),
+    );
+
+    const defaultVisibleColumns: string[] = initialColumnOrder
+        .filter((columnId) => !columns[columnId].isHiddenByDefault)
+        .map((columnId) => columnId.toString());
+    const [visibleColumns, setVisibleColumns] = useState(defaultVisibleColumns);
+
+    const renderCellValue = ({
+        rowIndex,
+        columnId,
+        setCellProps,
+    }: EuiDataGridCellValueElementProps) => {
+        const { pageSize, pageIndex } = pagination;
+        const rowIndexOnPage = rowIndex - pageIndex * pageSize;
+
+        const dataRow = data[rowIndexOnPage];
+        if (!dataRow) {
+            return;
+        }
+
+        const column = columns[columnId as keyof T];
+        const cellValue = dataRow[columnId as keyof T];
+
+        handleRowClick &&
```

Comment:
I would agree to be careful with truthy and falsy in case of values, however this one is to determine weather a the handleRowClick function is defined. Truthy and Falsy would be good enough here

Reply:
- I must stand my point
- Again. `&&` and `||` operators evaluate to strings and numbers and not `truthy` and `falsy` therefore and `if clause` without these operators

```
if (handleRowClick) {
    setCellProps({
        css: { cursor: 'pointer' },
        onClick: () => handleRowClick(dataRow),
    });
}
```